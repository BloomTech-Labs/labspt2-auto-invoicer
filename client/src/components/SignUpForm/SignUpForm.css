.signup-form {
  justify-content: center;
  display: flex;
  flex-wrap: wrap;
  padding: 20px 0;
  width: 600px;
  height: 100%;
}
.signup-form-group {
  margin: 10px auto;
  transition: ease-in-out color 0.2s;
}
.signup-form-group:hover {
  color: gray;
  cursor: pointer;
}

.signup-form-group input {
  border: 2px solid black;
  cursor: pointer;
  transition: ease-in-out border 0.3s;
  transition: ease-in-out box-shadow 0.2s;
}

.signup-form-group input:hover {
  border: 2px solid gray;
  box-shadow: 2px 2px 2px gray;
}

.signup-form-group input:focus {
  border: 2px solid blue;
  box-shadow: 2px 2px 2px blue;
}

.register-btn {
  min-width: 300px;
  margin: 20px auto 0;
  font-size: 2.4rem;
  border: 2px solid black;
}

.register-btn:hover {
  background: white;
  color: gray;
  border: 2px solid gray;
}
