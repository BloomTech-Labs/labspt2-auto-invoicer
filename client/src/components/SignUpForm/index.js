import React from "react";
import { connect } from "react-redux";
import { registerUser } from "../../actions/authActions";

// import icons
import EmailIcon from "@material-ui/icons/Email";
import PasswordIcon from "@material-ui/icons/LockOpen";
import Password2Icon from "@material-ui/icons/Lock";
import PhoneIcon from "@material-ui/icons/Phone";
import PersonIcon from "@material-ui/icons/AccountBox";
import Person2Icon from "@material-ui/icons/AssignmentInd";

// import css here
import "./SignUpForm.css";

class SignUpForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      password2: "",
      phone: "",
      firstName: "",
      lastName: ""
    };
  }
  changeHandler = e => {
    return this.setState({ [e.target.name]: e.target.value });
  };
  comparePasswords = (pass1, pass2) => {
    return pass1 === pass2 ? true : false;
  };
  createRegisterObject = e => {
    e.preventDefault();
    const {
      email,
      password,
      password2,
      phone,
      firstName,
      lastName
    } = this.state;
    // check if passwords match before moving forward
    if (this.comparePasswords(password, password2)) {
      if (firstName && lastName) {
        const name = `${firstName} ${lastName}`;
        const user = { email, password, phone, name };
        /* console.log until we have backend hooked up */
        console.log(user);
        this.props.registerUser(user);
        this.props.welcome(user);
      } else {
        return this.setState({ error: "Valid first/last name required" });
      }
    } else {
      return this.setState({
        error: "Passwords do not match"
      });
    }
  };
  render() {
    return (
      <form className="signup-form" onSubmit={this.createRegisterObject}>
        <label htmlFor="email" className="signup-form-group">
          <EmailIcon className="signup-icon" />
          <input
            className="signin-input"
            type="text"
            placeholder="email"
            name="email"
            id="email"
            onChange={this.changeHandler}
            value={this.state.email}
          />
        </label>
        <label htmlFor="firstName" className="signup-form-group">
          <PersonIcon className="signup-icon" />
          <input
            className="signin-input"
            type="text"
            placeholder="first name"
            name="firstName"
            id="firstName"
            onChange={this.changeHandler}
            value={this.state.firstName}
          />
        </label>
        <label htmlFor="lastName" className="signup-form-group">
          <Person2Icon className="signup-icon" />
          <input
            className="signin-input"
            type="text"
            placeholder="last name"
            name="lastName"
            id="lastName"
            onChange={this.changeHandler}
            value={this.state.lastName}
          />
        </label>
        <label htmlFor="phone" className="signup-form-group">
          <PhoneIcon className="signup-icon" />
          <input
            className="signin-input"
            type="text"
            placeholder="phone number"
            name="phone"
            id="phone"
            onChange={this.changeHandler}
            value={this.state.phone}
          />
        </label>
        <label htmlFor="password" className="signup-form-group">
          <PasswordIcon className="signup-icon" />
          <input
            className="signin-input"
            type="password"
            placeholder="password"
            name="password"
            id="password"
            onChange={this.changeHandler}
            value={this.state.password}
          />
        </label>
        <label htmlFor="password2" className="signup-form-group">
          <Password2Icon className="signup-icon" />
          <input
            className="signin-input"
            type="password"
            placeholder="password"
            name="password2"
            id="password2"
            onChange={this.changeHandler}
            value={this.state.password2}
          />
        </label>
        <button className="authentication-btns register-btn" type="submit">
          Register
        </button>
      </form>
    );
  }
}

const mapStateToProps = state => {
  return {};
};

export default connect(
  mapStateToProps,
  { registerUser }
)(SignUpForm);
