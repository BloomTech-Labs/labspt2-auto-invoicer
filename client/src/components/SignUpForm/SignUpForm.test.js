import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import { Provider } from "react-redux";
import SignUpForm from "../SignUpForm";
import { createStore } from "redux";
import rootReducer from "../../reducers/combineReducer";
const store = createStore(rootReducer);

describe("<SignUpForm />", () => {
  it("renders Sign Up Form Component", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SignUpForm />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
