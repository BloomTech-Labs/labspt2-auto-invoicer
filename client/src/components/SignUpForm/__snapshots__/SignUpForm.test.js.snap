// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> renders Sign Up Form Component 1`] = `
<DocumentFragment>
  <form
    class="signup-form"
  >
    <label
      class="signup-form-group"
      for="email"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1 signup-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <input
        class="signin-input"
        id="email"
        name="email"
        placeholder="email"
        type="text"
        value=""
      />
    </label>
    <label
      class="signup-form-group"
      for="firstName"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1 signup-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <input
        class="signin-input"
        id="firstName"
        name="firstName"
        placeholder="first name"
        type="text"
        value=""
      />
    </label>
    <label
      class="signup-form-group"
      for="lastName"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1 signup-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm0 4c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1.4c0-2 4-3.1 6-3.1s6 1.1 6 3.1V19z"
        />
      </svg>
      <input
        class="signin-input"
        id="lastName"
        name="lastName"
        placeholder="last name"
        type="text"
        value=""
      />
    </label>
    <label
      class="signup-form-group"
      for="phone"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1 signup-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
        />
      </svg>
      <input
        class="signin-input"
        id="phone"
        name="phone"
        placeholder="phone number"
        type="text"
        value=""
      />
    </label>
    <label
      class="signup-form-group"
      for="password"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1 signup-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6h1.9c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm0 12H6V10h12v10z"
        />
      </svg>
      <input
        class="signin-input"
        id="password"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </label>
    <label
      class="signup-form-group"
      for="password2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1 signup-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
        />
      </svg>
      <input
        class="signin-input"
        id="password2"
        name="password2"
        placeholder="password"
        type="password"
        value=""
      />
    </label>
    <button
      class="authentication-btns register-btn"
      type="submit"
    >
      Register
    </button>
  </form>
</DocumentFragment>
`;
